packages/orchestration/src/exos/agoric-names-tools.js

```diff
@@ -0,0 +1,66 @@
+import { VowShape } from '@agoric/vow';
+import { E, Far } from '@endo/far';
+import { M } from '@endo/patterns';
+import { BrandShape } from '@agoric/ertp';
+
+const { Fail } = assert;
+
+/**
+ * @import {NameHub} from '@agoric/vats';
+ * @import {AssetInfo} from '@agoric/vats/src/vat-bank';
+ * @import {Remote} from '@agoric/internal';
+ * @import {Vow, VowTools} from '@agoric/vow';
+ * @import {Zone} from '@agoric/zone';
+ */
+
+/**
+ * Perform remote calls to agoricNames in membrane-friendly way. This is an
+ * interim approach until https://github.com/Agoric/agoric-sdk/issues/9541,
+ * https://github.com/Agoric/agoric-sdk/pull/9322, or
+ * https://github.com/Agoric/agoric-sdk/pull/9519
+ *
+ * XXX consider exposing `has`, `entries`, `keys`, `values` from `NameHub`
+ *
+ * @param {Zone} zone
+ * @param {{ agoricNames: Remote<NameHub>; vowTools: VowTools }} powers
+ */
+export const makeAgNamesTools = (
+  zone,
+  { agoricNames, vowTools: { watch } },
+) => {
+  const agNamesTools = zone.exo(
+    'AgoricNamesTools',
+    M.interface('AgoricNamesToolsI', {
+      lookup: M.call().rest(M.arrayOf(M.string())).returns(VowShape),
+      findBrandInVBank: M.call(BrandShape).returns(VowShape),
+    }),
+    {
+      /** @param {...string} args */
+      lookup(...args) {
+        return watch(E(agoricNames).lookup(...args));
+      },
+      /**
+       * @param {Brand<'nat'>} brand
+       * @returns {Vow<AssetInfo>}
+       */
+      findBrandInVBank(brand) {
+        // XXX consider caching and refetching 1x if brand is not found,
+        // if this is planned to be local / long-lived
+        const vbankAssetNameHubP = E(agoricNames).lookup('vbankAsset');
+        return watch(
+          E(vbankAssetNameHubP).values(),
+          Far('BrandsWatcher', {
```

Comment:
`Far`? Is that how `watch` is supposed to be used? I don't see `Far` used in [watch.test.js](https://github.com/Agoric/agoric-sdk/blob/c61e75ca528c7ecdbe4663af6e95d78b4cb4ca5c/packages/vow/test/watch.test.js). I see exos there.

I don't see any docs either way on the `watch` function nor the `Watcher` type.